README.md

```diff
@@ -30,33 +35,60 @@ After that, you'll need to get the python files. If you've cloned this repo, the
 ```bash
 python3 path/to/plasmidi.py path/to/my_file.mid
 ```
+
 It will output a JSON string representation which you can either pipe into a program or file to parse it.
 
 If you only want one or more specific tracks, append the track number or name onto the command, and it will only output the tracks that exactly match those names/numbers:
+
 ```bash
 python3 path/to/plasmidi.py path/to/my_file.mid 3 "DRUMS"
 ```
 
 On Windows, you can also pipe the output into your clipboard like so:
+
 ```bash
 python3 path/to/plasmidi.py path/to/my_file.mid 3 "DRUMS" | clip.exe
 ```
 
-## Local Server
-If you want to contribute to the web part of the project, you'll need to run the server locally to test it. You'll need to have [Node.js](https://nodejs.org/) installed in order to run the server. Once you have it, enter the following commands to clone and move to the server, then install all the dependencies and launch it. 
+## Running Locally / Contributing
+
+To run the project locally, docker is strongly encouraged, but not directly required.
+
+### Wtih Docker
+
+**If you have docker, you can host a local version of appwrite to faclilty development. To get set up clone the repo and follow the next steps**
+
+1. Copy .env.example and create a .env file.
+2. Run `make appwrite` to initilize the appwrite containers. This will prompt you with some setup options. all defaults should be fine
+3. Head to http://localhost and create a local appwrite account and a project
+4. Create an api key in the project. to do this I went to settings -> view api keys -> add api key. It's recommended to provide it complete access locally.
+5. Open the .env file (listed in step 1) and fill in the top variables with your projects information. (Locally you can leave APPWRITE_URL as it is. It's using a docker network to directly connect the server to appwrite)
+6. Run `make up logs migrate`. This will do the following:
+   - start your server
+   - show you the logs for your server
+   - run the migration file to initilize your project
+
+Congrats! you're up and running
```

Comment:
Instructions for how to start the project. @JoePShoulak can you try this and see if you can follow along plz

Reply:
Just seeing this, will do tonight!

Reply:
@ebettenga 

1. I copied the .env and made no changes to it
1a. I got an error saying the Docker Daemon wasn't running. Launching the Docker app on Windows fixed that. 
2. This totally worked. 
3. I got stuck here for a second before I realized it wasn't on port 3000
4. I think I nailed this. 
5. Yup.
6. Seems like we may need to adjust the `package.json`(s) and maybe throw an `npm i` into the makefile somewhere for good measure. 
```shell
node:internal/modules/cjs/loader:1085
[0]   throw err;
[0]   ^
[0]
[0] Error: Cannot find module 'node-appwrite'
```